data/ha/hawk_test.py

```diff
@@ -0,0 +1,829 @@
+EDIT_MONITOR_TIMEOUT = "//*[@id=\"oplist\"]/fieldset/div/div[1]/div[3]/div[2]/div/div/a[1]"
+MODAL_MONITOR_TIMEOUT = "//*[@id=\"modal\"]/div/div/form/div[2]/fieldset/div/div[1]/div"
+TARGET_ROLE_STARTED = '//option[contains(@value, "tarted")]'
+HREF_DELETE_FORMAT = '//a[contains(@href, "%s") and contains(@title, "Delete")]'
+COMMIT_BTN_DANGER = '//button[contains(@class, "btn-danger") and contains(@class, "commit")]'
+CLONE_DATA_HELP_FILTER = '//a[contains(@data-help-filter, ".clone")]'
+OPT_STONITH = '//option[contains(@value, "stonith-sbd")]'
+RSC_OK_SUBMIT = '//input[contains(@class, "submit")]'
+GROUP_DATA_FILTER = '//a[contains(@data-help-filter, ".group")]'
+STONITH_CHKBOX = '//input[contains(@type, "checkbox") and contains(@value, "stonith-sbd")]'
+HREF_CONSTRAINTS = '//a[contains(@href, "#constraints")]'
+HREF_NODES = '//a[contains(@href, "#nodes")]'
+HREF_TAGS = '//a[contains(@href, "#tags")]'
+HREF_ALERTS = '//a[contains(@href, "#alerts")]'
+HREF_FENCING = '//a[contains(@href, "#fencing")]'
+RSC_ROWS = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr"
+DROP_DOWN_FORMAT = "//*[@id=\"resources\"]/div[1]/div[2]/div[2]/table/tbody/tr[%d]/td[6]/div/div"
+STONITH_MAINT_ON = '//a[contains(@href, "stonith-sbd/maintenance_on")]'
+NODE_DETAILS = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/a[2]"
+DISMISS_MODAL = "//*[@id=\"modal\"]/div/div/div[3]/button"
+CLEAR_STATE = "//*[@id=\"nodes\"]/div[1]/div[2]/div[2]/table/tbody/tr[1]/td[5]/div/div/button"
+NODE_MAINT = '//a[contains(@href, "maintenance") and contains(@title, "Switch to maintenance")]'
+NODE_READY = '//a[contains(@href, "ready") and contains(@title, "Switch to ready")]'
+GENERATE_REPORT = "//*[@id=\"generate\"]/form/div/div[2]/button"
+CLONE_CHILD = ('//select[contains(@data-help-filter, ".row.resource") and '
+               'contains(@name, "clone[child]")]')
+TARGET_ROLE_FORMAT = ('//select[contains(@class, "form-control") and '
+                      'contains(@name, "%s[meta][target-role]")]')
+STONITH_MAINT_OFF = ('//a[contains(@href, "stonith-sbd") and '
+                     'contains(@title, "Disable Maintenance Mode")]')
+
+# Functions
+
+### Functions to check types and classes
+
+def checkint(i):
+    if isinstance(i, int):
+        return i
+    else:
+        # Default to 1 if no int argument
+        return 1
+
+def checkstr(s):
+    if isinstance(s, str):
+        return s
+    else:
+        return str(s)
+
+def is_driver(d):
+    clase = str(type(d)).split(' ')[1][1:19]
+    if clase != 'selenium.webdriver':
+        print('ERROR: Driver must be of type webdriver. Got: [%s]' % type(d))
+        return False
+    return True
+
+def is_ssh(s):
+    clase = str(type(s)).split(' ')[1][1:26]
+    if clase != 'paramiko.client.SSHClient':
+        print('ERROR: SSH object must be of type paramiko.client.SSHClient. Got: [%s]' % type(s))
+        return False
+    return True
+
+### Logging and results
+
+def logresults(filename):
```

Comment:
why not to use something native ? somrthing like https://docs.python.org/3/library/logging.html ?

Reply:
Looks good. Don't think it's a good alternative for this function (as a `json.dumps()` of the `dict` is simpler), but could help elsewhere in the script. Will look into. (should've named this function `saveresults` instead of `logresults` to avoid confusion)